src/commands/api/deployImageApi/deployImageApi.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { callWithMaskHandling, callWithTelemetryAndErrorHandling, createSubscriptionContext, type IActionContext, type ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { ImageSource, acrDomain } from "../../../constants";
+import { getDomainFromRegistryName, getRegistryFromAcrName } from "../../../utils/imageNameUtils";
+import { pickContainer } from "../../../utils/pickItem/pickContainer";
+import { deployImage } from "../../deployImage/deployImage";
+import { type ContainerRegistryImageSourceContext } from "../../image/imageSource/containerRegistry/ContainerRegistryImageSourceContext";
+import { type DeployImageToAcaOptionsContract } from "../vscode-azurecontainerapps.api";
+
+export async function deployImageApi(deployImageOptions: DeployImageToAcaOptionsContract): Promise<void> {

```

Comment:
This is all a carryover of existing code.  Only difference is the `callWithTelemetryAndErrorHandling` wrap